Documentation/api-reference.md

```diff
@@ -68,72 +71,193 @@ m.aepSdk = AdobeAEPSDKInit()
+- `@return version as string`
 
 ##### Example
 
 ```brightscript
-adbEcidCallback = sub(context, ecid)
-  ' Handle the returned ECID value
-  print "getECID(): " + FormatJson(ecid)
-end sub
-
-m.aepSdk.getExperienceCloudId(adbEcidCallback, m)
+sdkVersion = m.aepSdk.getVersion()
 ```
+
 ---
 
-### getVersion
+### setLogLevel
 
 ##### Syntax
 
 ```brightscript
-getVersion: function() as string
+setLogLevel: function(level as integer) as void
 ```
 
-- `@return version as string`
+- `@param level as integer : the accepted values are (VERBOSE: 0, DEBUG: 1, INFO: 2, WARNING: 3, ERROR: 4)`
 
 ##### Example
 
 ```brightscript
-sdkVersion = m.aepSdk.getVersion()
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+m.aepSdk.setLogLevel(ADB_CONSTANTS.LOG_LEVEL.VERBOSE)
 ```
 
 ---
 
-### resetIdentities
+### shutdown
 
-Call this function to reset the Adobe identities such as ECID from the AEP Roku SDK.
+Call this function to shut down the AEP Roku SDK and drop further API calls.
 
 ##### Syntax
 
 ```brightscript
-resetIdentities: function() as void
+shutdown: function() as void
 ```
 
 ##### Example
 
 ```brightscript
-m.aepSdk.resetIdentities()
+m.aepSdk.shutdown();
 ```
 
 ---
 
+### updateConfiguration
+
+> **Note**
+> Some public APIs need valid configuration to process the data and make the network call to Adobe Experience Edge Network. All the hits will be queued if no valid configuration is found. It is ideal to call updateConfiguration API with valid require configuration before any other public APIs.
+
+#### Configuration Keys
+
+- Required for all APIs
+
+| Constants | Raw value | Type | Required |
+| :-- | :--: | :--: | :--: |
+| `ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID` | "edge.configId" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN` | "edge.domain" | String | **No**
+| `ADB_CONSTANTS.CONFIGURATION.CONSENT_DEFAULT` | "consent.default" | Map | **No**
+
+- Required for Media tracking APIs
+
+| Constants | Raw value | Type | Required |
+| :-- | :--: | :--: | :--: |
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_CHANNEL` | "edgemedia.channel" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_PLAYER_NAME` | "edgemedia.playerName" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_APP_VERSION` | "edgemedia.appVersion" | String | **No**
+
+##### Syntax
+
+```brightscript
+updateConfiguration: function(configuration as object) as void
+```
+
+- `@param configuration as object`
+
+##### Example
+
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID] = "<YOUR_CONFIG_ID>"
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN] = "<YOUR_DOMAIN_NAME>"
+
+' Set the default consent to pending and update the consent later based on user preference
```

Comment:
This example sets the default consent to pending. You can later update the consent based on user preferences using the setConsent API.